### ISSUE TITLE
[MU4 Issue] Chord symbols do not render in MusicXML files

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#13214]
**Describe the bug**
When loading a MusicXML file, any chord symbols present or loaded after do not render correctly - the roots are missing (so you see just "7" instead of "G7")

**To Reproduce**
1. Open any MusicXML file
2. If it has chord symbols, look at them
3. If not, try entering a new chord symbol: select first note, Ctrl+K, type "G7", hit Esc
4. See error: The chord symbols are missing their roots  "7" instead of "G7" etc.

**Expected behavior**
Chord symbols should display with their roots

**Screenshots**
![image](https://user-images.githubusercontent.com/1799936/188709321-2d605852-2fce-4736-accf-dd213f0649e7.png)

**Platform information**
 - OS: Linux

**Additional context**
The problem is that the chord description file that would normally be loaded as part of the basic score reading process seems to be skipped for MusicXML files.  The workaround is to go to Format / Style / Chord Symbols, change the appearance from Standard to Jazz and then back.  This forces the style file to be loaded.

This was originally reported in https://musescore.org/en/node/334422 but I just ran into it again and realized it was never reported here on GitHub.

**Comment by lvinken** [comment]
Issue does not reproduce for me using commit 56b4855 on x86_64 Ubuntu 22.04.

**Comment by edellaq** [comment]
On OSX it happens too.

My setup:
iMac (Retina 4K, 21.5-inch, Late 2015), 3,1 GHz Intel Core i5 quad-core
8 GB 1867 MHz DDR3, Intel Iris Pro Graphics 6200 1536 MB
OSX 12.6.2 (Monterey)

**Comment by MarcSabatella** [comment]
@Tantacrul this is coming up quite often and is presumably a very simple fix.

@lvinken - I don't think the issue is likely to be in the MusicXML code specifically but in how the initialization of the score style is handled. Somewhere there should be a call to function that loads a chord description file during import.  Unfortunately I don't really know how it is supposed to work anymore.

I've edited the description to include a MusicXML file that reproduces the problem.

**Comment by MarcSabatella** [comment]
That too; description updated.

**Comment by MarcSabatella** [comment]
As per newly added comments, this also affect MIDI.  Wherever the code to initiatialize the chordlist moved to, it isn't happening for imported scores.

**Comment by lvinken** [comment]
Will have a look at the MusicXML import code, but as far as I recall it does not explicitly load a score style.

**Comment by MarcSabatella** [comment]
Right, more likely it uses the Score constructor to create a score with the default style, and this formerly initialized the chordlist but for whatever reason doesn't anymore.

A little more browsing turned up this PR - https://github.com/musescore/MuseScore/issues/10103 - to fix a similar issue in "regular" (non-imported) scores.  I think the relevant code is the calls to ChordList::read() and to ChordList::checkChordList().  Not sure if it makes sense to factor those out somewhere so each importer doesn't have to do this explicitly.  @RomanPudashkin , do you have thoughts on this?

**Comment by Jojo-Schmitz** [comment]
Remark of @laturetab on https://musescore.org/en/node/342157#comment-1171128:
```
Take a look at harmony.cpp on lines 799 and 806. Change variable "s" for "ss" on both these lines and musicxml, guitarpro,... chords will magically appear after import.
```

https://github.com/musescore/MuseScore/blob/6658b813e832e8deba829f9316269404acb5a966/src/engraving/libmscore/harmony.cpp#L799-L806

**Comment by MarcSabatella** [comment]
Without understanding more about the proposed change, I can't see why it would fix anything and more importantly, why it wouldn't be expected to break anything.  That code hasn't changed since MuseScore 3 where it worked correctly.

**Comment by laturetab** [comment]
Marc -- For the chord "G", at line 799 the value of "s" is empty. A few lines above this: s = s.mid(idx) where idx==1 and s=="G". So _parsedForm->parse is parsing "" and descr() is looking up "". Changing "s" to "ss" makes it parse() and desc() the original string "G".

So, we either don't do s = s.mid() or we change "s" to "ss".

The change I suggested does work however s=s.mid(idx) is the true Dr Evil.

Maybe the solution is to do something like chordType = s.mid(idx) and leave "s" alone.

Hopefully this inspires someone to take a look.

**Comment by Jojo-Schmitz** [comment]
That `s = s.mid(idx)` is there since 2015-09-14 !

**Comment by cbjeukendrup** [comment]
Might it be caused by the fact that `QString` has been replaced with our own `String`? Git Blame specifically mentions https://github.com/musescore/MuseScore/commit/c31e7fb0720817a4a1558c02ca17f71dc941b7a6 somewhere.

**Comment by MarcSabatella** [comment]
I am pretty sure this is all a red herring.

To be clear - s is *supposed* to be empty at that point.  It's not parsing the *root*, it's parsing the *rest of the chord*.  So if that had been G7b9, it is supposed to parse the "7b9" only.  The root was already parsed and removed - that's the whole point of the "mid" call, to remove the root from the equation.

So as far as I can tell replacing s with ss would actually *break* the parsing, forcing the parser to read the "G" as if were not the root.  The results might look good for a moment if there is nothing else in the chord symbol to parse (I.e, if this is just a major triad) but would be wrong going forward - the rest of the chord would not parse correctly.

The problem is simply with the style information as I observed earlier.  There is a missing call to the chordlist initialization.  This is easily confirmed by forcing the chordlist to be re-initialized by changing style to Jazz and back to Standard.

**Comment by Tantacrul** [comment]
@laturetab - that sounds like a different issue so it would be great if you could log it separately. Thanks!

**Comment by laturetab** [comment]
I confirmed what Marc mentioned above about loading the chordlist file. The loading of the chordlist file happens when you open an mscz file (not other file types.) So if you open song.mscz, the chordlist will get loaded and everything works, including palette fretboards. If you don't load an mscz (e.g. guitarpro, musicxml, midi, ...) chord names will get butchered in the score and on the fret board diagrams.

Shouldn't the chordlist be loaded only once and used by everything? It seems like a complete waste of time to load this over and over and over ...

**Comment by laturetab** [comment]
This is my last comment on this issue :-)

score->chordList()->read(u"chords_std.xml");

Add this to the start of every file importer (just after you open the import file). I found this in iex_bb. It needs to be "chords_std.xml" and not "chords.xml"

Fixes everything including fretboards. Don't have to touch harmony.cpp :-)

Sorry for all the confusion.

**Comment by MarcSabatella** [comment]
The chordlist is per score, because different scores can use different chord styles.

It would definitely be possible to fix this with a manual read of the chordlist into each importer (and I think there is a better way than hardcoding the pathname every time).  But, that seems wrong.  The overall style is already being initialiezd; the chord initialization should simply happen there when the score style is initialized.  That's how it used to be but was factored out for some reason.  It should be refactored again so that all importers (present and future) get this for free.
